### ISSUE TITLE
Incorrect gradient returned when vector splatting is used

### ISSUE THREAD

**Issue by GiggleLiu** [EnzymeAD/Enzyme.jl#545]
I have a simple program that tries to find a unit disk embedding of a graph. When I use vector splatting in the `getloc` function, the gradient returned by `Enzyme` is incomplete. 

```julia
using Graphs, Enzyme

function unitdisk_loss(graph::SimpleGraph, locs::AbstractMatrix)
    n = nv(graph)
    loss = 0.0
    @assert size(locs, 2) == n-1 && size(locs, 1) == 2  # fix one node
    for i=1:n
        x, y = getloc(locs, i)
        for j=i+1:n
            u, v = getloc(locs, j)
            d2 = (x-u)^2 + (y-v)^2
            if has_edge(graph, i, j)
                if d2 > 0.99   # must be smaller than 1.0
                    loss += d2 - 0.99
                end
            else
                if d2 < 1.01   # must be smaller than 1.0
                    loss += 1.01 - d2
                end
            end
        end
    end
    return loss
end

function getloc(locs, i)
    if i==1  # fix one of the nodes
        x, y = 0.0, 0.0
    else
	# correct
        x, y = locs[1,i-1], locs[2,i-1]
	# incorrect
	#x, y = locs[:,i-1]
    end
    return x, y
end

let
    graph = smallgraph(:petersen)
    x0 = ones(2, 9)
    din = zeros(2, 9)
    Enzyme.autodiff(unitdisk_loss, Const(graph), Duplicated(x0, din))
    din
end
```

## Output
```
# incorrect output
2×9 Matrix{Float64}:
 2.0  0.0  0.0  2.0  2.0  0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0

# correct output
2×9 Matrix{Float64}:
 2.0  0.0  0.0  2.0  2.0  0.0  0.0  0.0  0.0
 2.0  0.0  0.0  2.0  2.0  0.0  0.0  0.0  0.0
```

## Version Info
```
(@v1.8) pkg> st Enzyme
Status `~/.julia/environments/v1.8/Project.toml`
  [7da242da] Enzyme v0.10.12
```

BTW, the speed of Enzyme is incrediblly fast.

**Comment by tgymnich** [comment]
The given code snippet produces the correct output for me using Julia 1.8.3 and Enzyme v0.10.12 on my M1 Mac.

**Comment by GiggleLiu** [comment]
My system is Ubuntu 22.04 x64. Julia 1.8.3 and Enzyme v0.10.12.

@tgymnich have you uncommented the line below `# incorrect`?

**Comment by tgymnich** [comment]
Now I can reproduce it. Thanks!

**Comment by tgymnich** [comment]
Reduce test case:
```
using Enzyme

@inline function getloc(locs, i)
	# correct
    #    x, y = @inbounds locs[1,i], locs[2,i]
	# incorrect
	    x, y = @inbounds locs[:,i-1]
    return x, y
end

function test(mat,i)
    loss = 0.0
    _,y = getloc(mat,i)
    loss += y
    return loss
end

x0 = ones(2, 9)
din = zeros(2, 9)
Enzyme.autodiff(test, Duplicated(x0, din), Const(1))
Enzyme.Compiler.enzyme_code_llvm(test, Const, Tuple{Duplicated{Matrix{Float64}}, Const{Int64}}, run_enzyme=false)
Enzyme.Compiler.enzyme_code_llvm(test, Const, Tuple{Duplicated{Matrix{Float64}}, Const{Int64}}, run_enzyme=true)
display(din)
```
